source/Logger.cpp

```diff
@@ -0,0 +1,42 @@
+/* Logger.cpp
+Copyright (c) 2022 by Peter van der Meer
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+*/
+
+#include "Logger.h"
+
+#include <iostream>
+#include <mutex>
+
+using namespace std;
+
+namespace {
+	function<void(const string &message)> logErrorCallback = nullptr;
+	mutex logErrorMutex;
+}
+
+
+
+void Logger::SetLogErrorCallback(function<void(const string &message)> callback)
+{
+	logErrorCallback = callback;
+}
+
+
+
+void Logger::LogError(const string &message)
+{
+	lock_guard<mutex> lock(logErrorMutex);
+	// Log by default to stderr.
+	cerr << message << endl;
```

Comment:
Should this be done inside the callback instead? That way, for unit tests, we can suppress the output without having that weird sink class.

Reply:
Yes, good idea. I moved it to the callback.

Reply:
I reverted this change for now; both the main-game as well as the unit-tests actually expect to have those messages sent to the output.

We can always update this in the future (it is a 6 lines change), but for this PR i prefer to just keep logging to stderr enabled by default for every class using the logger. (Since all users currently seem to expect output on stderr.)